app/src/main/java/com/academy/ramon/marvelcomicviewer/components/RetrofitModule.java

```diff
@@ -0,0 +1,36 @@
+package com.academy.ramon.marvelcomicviewer.components;
+
+import com.academy.ramon.marvelcomicviewer.api.MarvelAPI;
+
+import javax.inject.Singleton;
+
+import dagger.Module;
+import dagger.Provides;
+import retrofit2.Retrofit;
+import retrofit2.adapter.rxjava.RxJavaCallAdapterFactory;
+import retrofit2.converter.gson.GsonConverterFactory;
+
+/**
+ * Created by Ramon on 1/3/2017.
+ */
+
+@Module
+public class RetrofitModule {
+
+    @Provides
+    @Singleton
+    Retrofit provideRetrofit() {
+        return new Retrofit.Builder()
+                .addCallAdapterFactory(RxJavaCallAdapterFactory.create())
+                .addConverterFactory(GsonConverterFactory.create())
+                .baseUrl(MarvelAPI.ENDPOINT)
+                .build();
+    }
+
+    @Provides
+    @Singleton
+    MarvelAPI provideMarvelAPI() {
+        Retrofit retrofit = provideRetrofit();
```

Comment:
@Mathreyu since you are already providing a `Retrofit` instance on line `22` you are able to do `MarvelAPI provideMarvelAPI(Retrofit retrofit) {` and use that parameter to create your `MarvelAPI` instance. So you can delete this line, because if you call `provideRetrofit();` directly you are creating a new instance and dagger can not manage the singleton as you specified.